packages/main/src/TabContainer.js

```diff
@@ -500,16 +500,37 @@ class TabContainer extends UI5Element {
 		}
 
 		if (isEnter(event)) {
-			this._onHeaderItemSelect(tab);
+			event.preventDefault();
+			if (tab._realTab.isSingleClickArea) {
+				this._onTabStripClick(event);
+			} else {
+				this._onHeaderItemSelect(tab);
+			}
 		}
 
 		// Prevent Scrolling
 		if (isSpace(event)) {
 			event.preventDefault();
+			if (tab._realTab.isSingleClickArea) {
+				this._onTabStripClick(event);
+			}
+			walk(this.items, item => {
+				item._realTab = tab._realTab;
+			});
```

Comment:
Is it possible to ensure that the items will have their real tab set correctly before we even get to this function?